cardea/problem_definition/readmission.py

```diff
@@ -0,0 +1,144 @@
+        cutoff_time_label: The cutoff time label of the prediction problem
+        cutoff_entity: The entity name in which it contains the cutoff time label.
+        prediction_type: The type of the machine learning prediction.
+        """
+
+    updated_es = None
+    target_label = 'readmitted'
+    target_entity = 'Encounter'
+    cutoff_time_label = 'end'
+    cutoff_entity = 'Period'
+    prediction_type = 'classification'
+
+    def generate_cutoff_times(self, es):
+        """Generates cutoff times for the predection problem.
+
+            Args:
+            es: fhir entityset.
+
+            Returns:
+            es, target_entity, Series of target_label and a dataframe of cutoff_times
+
+            Raises:
+            ValueError: An error occurs if the cutoff variable does not exist.
+        """
+        try:
+            updated_es = self.generate_target_label(es)
+
+            self.check_target_label(
+                es,
+                self.cutoff_entity,
+                self.cutoff_time_label)  # check the existance of the cutoff label
+
+            cutoff_times = es[self.cutoff_entity].df[self.cutoff_time_label]
+            cutoff_times = cutoff_times.to_frame()
+            cutoff_times.index = es[self.cutoff_entity].df.index
+            cutoff_times = cutoff_times.rename(columns={self.cutoff_time_label:
+                                                        'cutoff_time'})
+
+            return (es, self.target_entity,
+                    updated_es[self.target_entity].df[self.target_label],
+                    cutoff_times)
+        except ValueError:
+            raise ValueError('Cutoff time label {} in table {} does not exist'
+                             .format(self.cutoff_time_label, self.target_entity))
+
+    def generate_target_label(self, es):
+        """Generates target labels in the case of having missing label in the entityset.
+
+            Args:
+            es: fhir entityset.
+            target_label: The target label of the prediction problem.
+            target_entity: The entity name in which it contains the target label.
+
+            Returns:
+            Updated entityset with the generated label.
+
+            Raises:
+            ValueError: An error occurs if the target label cannot be generated.
+            """
+        generate_from = 'Period'
+        end = 'end'
+
+        try:
+            if self.check_target_label(
+                es,
+                generate_from,
+                end) and self.check_target_label(es,
+                                                 self.target_entity,
+                                                 'period'):
+
+                try:
+                    self.check_target_label_values(
+                        es,
+                        generate_from,
+                        end)
+
+                    entity_set_df = es[self.target_entity].df
+                    generate_df = es[generate_from].df
```

Comment:
I recommend we change the variable name to make it more clear:
```suggestion
                    generated_df = es[generate_from].df
                    merged_df = pd.merge(entity_set_df, generated_df,
```